docs/book/application-integration/usage-in-a-laminas-mvc-application.md

```diff
@@ -0,0 +1,181 @@
+# Usage in a laminas-mvc Application
+
+The following example shows _one_ potential use case of laminas-form within
+a laminas-mvc based application. The example uses a module, a controller and the
+laminas-form element manager.
+
+The example is based on the [tutorial application](https://docs.laminas.dev/tutorials/getting-started/overview/)
+which builds an album inventory system.
+
+Before starting, make sure laminas-form is installed and configured.
+
+## Create Form
+
+Create a form as separate class, e.g. `module/Album/src/Form/AlbumForm.php`:
+
+```php
+namespace Album\Form;
+
+use Laminas\Form\Element\Text;
+use Laminas\Form\Form;
+
+class AlbumForm extends Form
+{
+    public function init()
+    {
+        // Title
+        $this->add(
+            [
+                'name'    => 'title',
+                'type'    => Text:class,
+                'options' => [
+                    'label' => 'Title',
+                ],
+            ]
+        );
+    
+        // …
+    }
+}
+```
+
+## Using Form
+
+### Create Controller
+
+Using the form in a controller, e.g.
+`module/Album/Controller/AlbumController.php`:
+
+```php
+namespace Album\Controller;
+
+use Album\Form\AlbumForm;
+use Laminas\Form\FormInterface;
+use Laminas\Mvc\Controller\AbstractActionController;
+
+class AlbumController extends AbstractActionController
+{
+    /** @var FormInterface */
+    private $form;
+    
+    public function __construct(FormInterface $form)
+    {
+        $this->form = $form;
+    }
+    
+    public function addAction()
+    {
+        // Set action attribute
+        $this->form->setAttribute(
+            'action',
+            $this->url()->fromRoute('album', ['action' => 'add'])
+        );
+
+        $variables = ['form' => $this->form];
+        
+        if (! $this->getRequest()->isPost()) {
+            return $variables;
+        }
+
+        // Validation
+        $this->form->setData($this->getRequest()->getPost());
+        if (! $this->form->isValid()) {
+            return $variables;
+        }
+    
+        // …
+
+        return $this->redirect()->toRoute('album', ['action' => 'add']);
+    }
+}
+```
+
+### Create Factory for Controller
+
+Fetch the `AlbumForm` from the form element manager in a factory,
+e.g. `src/Album/Controller/AlbumControllerFactory.php`:
+
+```php
+namespace Album\Controller;
+
+use Album\Form\AlbumForm;
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use Laminas\ServiceManager\PluginManagerInterface;
+
+class AlbumControllerFactory implements FactoryInterface
```

Comment:
I am not sure if we have any convention, but I would go with a bit more generic case:
_invokable factory without the interface_. Then we can use PSR-11 container interface, and we do not need two other arguments in invoke (as these are unused anyway).

Reply:
> I am not sure if we have any convention…

Nothing is written down, so no.